node/network/src/proxy/predicate.rs

```diff
@@ -0,0 +1,26 @@
+use futures::Stream;
+
+use crate::protocol::SimplePackedMessage;
+use crate::proxy::ProxyHandler;
+
+pub struct FnProxyHandler {
+    predicate: fn(SimplePackedMessage) -> Option<SimplePackedMessage>
```

Comment:
Trait `Fn` is not to be confused with 'fn': https://stackoverflow.com/questions/54181816/function-pointer-vs-fn-trait-object

Also prefer `FnMut` when possible.
```suggestion
    predicate: FnMut(SimplePackedMessage) -> Option<SimplePackedMessage>
```

Reply:
Moved to #803